app/controllers/api/v1/subjects_controller.rb

```diff
@@ -2,7 +2,9 @@ module API
   module V1
     class SubjectsController < API::V1::ApplicationController
       def index
-        @subjects = UCASSubject.all
+        @subjects = Subject.where.not(type: "DiscontinuedSubject")
+          .where.not(subject_code: nil)
```

Comment:
I think we need a scope for these to `where` clauses, they're used below a whole lot.